core/controllers/acl_decorators.py

```diff
@@ -487,6 +487,44 @@ def test_can_manage_account(self, **kwargs):
     return test_can_manage_account
 
 
+def can_manage_own_profile(handler):
+    """Decorator to check whether user can manage their profile.
```

Comment:
Suggest also including some brief details about what this does in the event the user cannot manage their profile (to provide context on how this should be used). This doesn't need to be as specific as the exceptions thrown, but indicating an exception will be thrown is really pertinent to users of the decorator.

Reply:
Does this seem okay?

```python
def can_manage_own_profile(handler):
    """Decorator to check whether a user can manage their profile. Only
    the particular profile user and full user of the account have access
    to this. Error will be thrown for others.
```

Reply:
Yup, that sounds clearer.

Reply:
Done.